packages/atomic/src/components/common/atomic-modal/atomic-modal.tsx

```diff
@@ -155,7 +161,18 @@ export class AtomicModal implements InitializableComponent<AnyBindings> {
                 part="body-wrapper"
                 class="overflow-auto grow flex flex-col items-center w-full"
               >
-                <div part="body" class="w-full max-w-lg">
+                <div
+                  part="body"
+                  class="w-full max-w-lg"
+                  ref={(element) =>
+                    element &&
+                    element.addEventListener(
```

Comment:
The listener should be automatically removed when there's no more references to the element.